### ISSUE TITLE
Configuring BASE_PATH does not works

### ISSUE THREAD

**Issue by Kani999** [netbox-community/netbox-docker#650]
### Current Behavior

After setting the base URL path to `BASE_PATH = 'test/'` the error about Static Media Failure is raised. 

![image](https://user-images.githubusercontent.com/17337078/141251361-c7297060-6129-48b6-847c-9d23b8acfd69.png)

### Expected Behavior

- Netbox app should run without a problem 
- Static files should be referenced according to `BASE_PATH`. 

### Docker Compose Version

1.29.2

### Docker Version

```text
Client: Docker Engine - Community
 Cloud integration: 1.0.17
 Version:           20.10.8
 API version:       1.41
 Go version:        go1.16.6
 Git commit:        3967b7d
 Built:             Fri Jul 30 19:54:22 2021
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.8
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.16.6
  Git commit:       75249d8
  Built:            Fri Jul 30 19:52:31 2021
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.4.9
  GitCommit:        e25210fe30a0a703442421b0f60afac609f950a3
 runc:
  Version:          1.0.1
  GitCommit:        v1.0.1-0-g4144b63
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

### The git Revision

6a52a48b71187eabb16db2149060cffc159f5f72

### The git Status

```text
On branch release
Your branch is up to date with 'origin/release'.
```

### Startup Command

docker compose up

### NetBox Logs

```text
Starting netbox-docker_redis-cache_1 ... done
Starting netbox-docker_postgres_1    ... done
Starting netbox-docker_redis_1       ... done
Starting netbox-docker_netbox-housekeeping_1 ... done
Starting netbox-docker_netbox-worker_1       ... done
Starting netbox-docker_netbox_1              ... done
Attaching to netbox-docker_postgres_1, netbox-docker_redis-cache_1, netbox-docker_redis_1, netbox-docker_netbox-worker_1, netbox-docker_netbox-housekeeping_1, netbox-docker_netbox_1
netbox-housekeeping_1  | Interval set to 86400 seconds
netbox-worker_1        | 🧬 loaded config '/etc/netbox/config/configuration.py'
netbox-housekeeping_1  | 🧬 loaded config '/etc/netbox/config/configuration.py'
netbox-worker_1        | 🧬 loaded config '/etc/netbox/config/extra.py'
netbox-worker_1        | 🧬 loaded config '/etc/netbox/config/logging.py'
netbox-housekeeping_1  | 🧬 loaded config '/etc/netbox/config/extra.py'
netbox-worker_1        | 🧬 loaded config '/etc/netbox/config/plugins.py'
netbox-housekeeping_1  | 🧬 loaded config '/etc/netbox/config/logging.py'
netbox-housekeeping_1  | 🧬 loaded config '/etc/netbox/config/plugins.py'
netbox-housekeeping_1  | [*] Clearing expired authentication sessions
netbox-housekeeping_1  |        Sessions cleared.
netbox-housekeeping_1  | [*] Checking for expired changelog records
netbox-housekeeping_1  |        No expired records found.
netbox-housekeeping_1  | [*] Checking for latest release
netbox-housekeeping_1  |        Latest release: 3.0.9
netbox-housekeeping_1  | Finished.
netbox_1               | 🧬 loaded config '/etc/netbox/config/configuration.py'
netbox_1               | 🧬 loaded config '/etc/netbox/config/extra.py'
netbox_1               | 🧬 loaded config '/etc/netbox/config/logging.py'
netbox_1               | 🧬 loaded config '/etc/netbox/config/plugins.py'
netbox_1               | 🧬 loaded config '/etc/netbox/config/configuration.py'
netbox_1               | 🧬 loaded config '/etc/netbox/config/extra.py'
netbox_1               | 🧬 loaded config '/etc/netbox/config/logging.py'
netbox_1               | 🧬 loaded config '/etc/netbox/config/plugins.py'
netbox_1               | ▶️  Running the startup script /opt/netbox/startup_scripts/000_users.py
netbox_1               | ▶️  Running the startup script /opt/netbox/startup_scripts/010_groups.py
netbox_1               | ▶️  Running the startup script /opt/netbox/startup_scripts/015_object_permissions.py
netbox_1               | ▶️  Running the startup script /opt/netbox/startup_scripts/020_custom_fields.py
netbox_1               | ▶️  Running the startup script /opt/netbox/startup_scripts/020_tags.py
netbox_1               | ▶️  Running the startup script /opt/netbox/startup_scripts/020_tenant_groups.py
netbox_1               | ▶️  Running the startup script /opt/netbox/startup_scripts/030_regions.py
netbox_1               | ▶️  Running the startup script /opt/netbox/startup_scripts/030_tenants.py
netbox_1               | ▶️  Running the startup script /opt/netbox/startup_scripts/040_sites.py
netbox_1               | ▶️  Running the startup script /opt/netbox/startup_scripts/050_manufacturers.py
netbox_1               | ▶️  Running the startup script /opt/netbox/startup_scripts/060_device_types.py
netbox_1               | ▶️  Running the startup script /opt/netbox/startup_scripts/070_rack_roles.py
netbox_1               | ▶️  Running the startup script /opt/netbox/startup_scripts/075_locations.py
netbox_1               | ▶️  Running the startup script /opt/netbox/startup_scripts/080_racks.py
netbox_1               | ▶️  Running the startup script /opt/netbox/startup_scripts/090_device_roles.py
netbox_1               | ▶️  Running the startup script /opt/netbox/startup_scripts/100_platforms.py
netbox_1               | ▶️  Running the startup script /opt/netbox/startup_scripts/130_cluster_types.py
netbox_1               | ▶️  Running the startup script /opt/netbox/startup_scripts/135_cluster_groups.py
netbox_1               | ▶️  Running the startup script /opt/netbox/startup_scripts/135_clusters.py
netbox_1               | ▶️  Running the startup script /opt/netbox/startup_scripts/140_clusters.py
netbox_1               | ▶️  Running the startup script /opt/netbox/startup_scripts/140_devices.py
netbox_1               | ▶️  Running the startup script /opt/netbox/startup_scripts/150_rirs.py
netbox_1               | ▶️  Running the startup script /opt/netbox/startup_scripts/160_aggregates.py
netbox_1               | ▶️  Running the startup script /opt/netbox/startup_scripts/165_cluster_groups.py
netbox_1               | ▶️  Running the startup script /opt/netbox/startup_scripts/175_route_targets.py
netbox_1               | ▶️  Running the startup script /opt/netbox/startup_scripts/180_vrfs.py
netbox_1               | ▶️  Running the startup script /opt/netbox/startup_scripts/190_prefix_vlan_roles.py
netbox_1               | ▶️  Running the startup script /opt/netbox/startup_scripts/200_vlan_groups.py
netbox_1               | ▶️  Running the startup script /opt/netbox/startup_scripts/210_vlans.py
netbox_1               | ▶️  Running the startup script /opt/netbox/startup_scripts/220_prefixes.py
netbox_1               | ▶️  Running the startup script /opt/netbox/startup_scripts/230_virtual_machines.py
netbox_1               | ▶️  Running the startup script /opt/netbox/startup_scripts/240_virtualization_interfaces.py
netbox_1               | ▶️  Running the startup script /opt/netbox/startup_scripts/250_dcim_interfaces.py
netbox_1               | ▶️  Running the startup script /opt/netbox/startup_scripts/260_ip_addresses.py
netbox_1               | ▶️  Running the startup script /opt/netbox/startup_scripts/270_primary_ips.py
netbox_1               | ▶️  Running the startup script /opt/netbox/startup_scripts/280_custom_links.py
netbox_1               | ▶️  Running the startup script /opt/netbox/startup_scripts/280_providers.py
netbox_1               | ▶️  Running the startup script /opt/netbox/startup_scripts/290_circuit_types.py
netbox_1               | ▶️  Running the startup script /opt/netbox/startup_scripts/290_webhooks.py
netbox_1               | ▶️  Running the startup script /opt/netbox/startup_scripts/300_circuits.py
netbox_1               | ▶️  Running the startup script /opt/netbox/startup_scripts/320_services.py
netbox_1               | ▶️  Running the startup script /opt/netbox/startup_scripts/330_power_panels.py
netbox_1               | ▶️  Running the startup script /opt/netbox/startup_scripts/340_power_feeds.py
netbox_1               | ✅ Initialisation is done.
netbox_1               | 2021/11/11 06:45:35 [warn] 7#7 Unit is running unprivileged, then it cannot use arbitrary user and group.
netbox_1               | 2021/11/11 06:45:35 [info] 7#7 unit started
netbox_1               | 2021/11/11 06:45:35 [info] 16#16 discovery started
netbox_1               | 2021/11/11 06:45:35 [notice] 16#16 module: python 3.9.5 "/usr/lib/unit/modules/python3.unit.so"
netbox_1               | 2021/11/11 06:45:35 [info] 7#7 controller started
netbox_1               | 2021/11/11 06:45:35 [notice] 7#7 process 16 exited with code 0
netbox_1               | 2021/11/11 06:45:35 [info] 18#18 router started
netbox_1               | 2021/11/11 06:45:35 [info] 18#18 OpenSSL 1.1.1l  24 Aug 2021, 101010cf
netbox_1               | 2021/11/11 06:45:35 [info] 19#19 "netbox" application started
netbox_1               | 🧬 loaded config '/etc/netbox/config/configuration.py'
netbox_1               | 🧬 loaded config '/etc/netbox/config/extra.py'
netbox_1               | 🧬 loaded config '/etc/netbox/config/logging.py'
netbox_1               | 🧬 loaded config '/etc/netbox/config/plugins.py'
netbox_1               | ⚙️ Applying configuration from /etc/unit/nginx-unit.json
netbox_1               | ✅ Unit configuration loaded successfully
netbox_1               | 2021/11/11 06:45:38 [notice] 7#7 process 14 exited with code 0
netbox_1               | 2021/11/11 06:46:49 [info] 34#34 "netbox" application started
netbox_1               | 🧬 loaded config '/etc/netbox/config/configuration.py'
netbox_1               | 🧬 loaded config '/etc/netbox/config/extra.py'
netbox_1               | 🧬 loaded config '/etc/netbox/config/logging.py'
netbox_1               | 🧬 loaded config '/etc/netbox/config/plugins.py'


netbox_1               | 172.22.0.1 - - [11/Nov/2021:06:46:53 +0000] "GET /test/ HTTP/1.1" 200 133840 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/95.0.4638.69 Safari/537.36"
netbox_1               | Not Found: /test/static/netbox-external.css
netbox_1               | 172.22.0.1 - - [11/Nov/2021:06:46:53 +0000] "GET /test/static/netbox-external.css HTTP/1.1" 404 7469 "http://localhost:8000/test/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/95.0.4638.69 Safari/537.36"
netbox_1               | Not Found: /test/static/netbox-light.css
netbox_1               | 172.22.0.1 - - [11/Nov/2021:06:46:53 +0000] "GET /test/static/netbox-light.css HTTP/1.1" 404 7460 "http://localhost:8000/test/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/95.0.4638.69 Safari/537.36"
netbox_1               | 2021/11/11 06:46:53 [info] 35#35 "netbox" application started
netbox_1               | Not Found: /test/static/netbox.js
netbox_1               | 172.22.0.1 - - [11/Nov/2021:06:46:53 +0000] "GET /test/static/netbox.js HTTP/1.1" 404 7439 "http://localhost:8000/test/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/95.0.4638.69 Safari/537.36"
netbox_1               | 172.22.0.1 - - [11/Nov/2021:06:46:54 +0000] "GET /test/media-failure/?filename=netbox-light.css HTTP/1.1" 200 1836 "http://localhost:8000/test/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/95.0.4638.69 Safari/537.36"
netbox_1               | Not Found: /test/static/netbox_logo.svg
netbox_1               | 172.22.0.1 - - [11/Nov/2021:06:46:54 +0000] "GET /test/static/netbox_logo.svg HTTP/1.1" 404 7457 "http://localhost:8000/test/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/95.0.4638.69 Safari/537.36"
netbox_1               | 🧬 loaded config '/etc/netbox/config/configuration.py'
netbox_1               | 🧬 loaded config '/etc/netbox/config/extra.py'
netbox_1               | 🧬 loaded config '/etc/netbox/config/logging.py'
netbox_1               | 🧬 loaded config '/etc/netbox/config/plugins.py'
netbox_1               | Not Found: /favicon.ico
netbox_1               | 172.22.0.1 - - [11/Nov/2021:06:46:54 +0000] "GET /favicon.ico HTTP/1.1" 404 2109 "http://localhost:8000/test/media-failure/?filename=netbox-light.css" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/95.0.4638.69 Safari/537.36"
netbox_1               | Not Found: /test/static/netbox-dark.css
netbox_1               | 172.22.0.1 - - [11/Nov/2021:06:46:56 +0000] "GET /test/static/netbox-dark.css HTTP/1.1" 404 7457 "http://localhost:8000/test/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/95.0.4638.69 Safari/537.36"
netbox_1               | 2021/11/11 06:48:57 [notice] 7#7 process 34 exited with code 0
netbox_1               | 2021/11/11 06:48:57 [notice] 7#7 process 35 exited with code 0
```

### Content of docker-compose.override.yml

```yaml
version: '3.4'
services:
  netbox:
    ports:
      - 8000:8080
```

**Comment by Kani999** [comment]
I've just pulled the `netbox-docker` and tried to do the fresh start with `BASE_PATH` and the problem still occurs.

**Comment by kale1d0code** [comment]
I did get this working about a month ago by setting the environment variable BASE_DIR=/opt/netbox/netbox

not sure what has happened since then as I am now getting the static media failure message again.
it would be better to take nginx out of the container completely and have it run on fpm/fcgi

**Comment by kale1d0code** [comment]
I have attempted setting
the following environment variables
```
BASE_PATH=netbox
BASE_PATH=netbox/
BASEDIR=/opt/netbox
BASE_DIR=/opt/netbox
BASEDIR=/opt/netbox/netbox
BASE_DIR=/opt/netbox/netbox
STATIC_ROOT=/opt/netbox/netbox/static
STATIC_URL=/netbox/static
SKIP_STARTUP_SCRIPTS=true
```

**Comment by kkthxbye-code** [comment]
The issue is that the match pattern configured in nginx unit for static files is static (as in, non-configurable):

https://github.com/netbox-community/netbox-docker/blob/release/docker/nginx-unit.json#L10-L16

You might be able to mount your own nginx-unit.json file to /etc/unit/nginx-unit.json with a correct route for static files.

**Comment by kale1d0code** [comment]
A good fix might be to have this file generated when the container starts

**Comment by kale1d0code** [comment]
> The issue is that the match pattern configured in nginx unit for static files is static (as in, non-configurable):
> 
> https://github.com/netbox-community/netbox-docker/blob/release/docker/nginx-unit.json#L10-L16
> 
> You might be able to mount your own nginx-unit.json file to /etc/unit/nginx-unit.json with a correct route for static files.

can you provide an example?
I've tried editing those lines to
```
    {
      "match": {
        "uri": "/netbox/static/*"
      },
      "action": {
        "share": "/opt/netbox/netbox"
      }
    },
```

I have also tried
```
    {
      "match": {
        "uri": "/netbox/*"
      },
      "action": {
        "share": "/opt/netbox/netbox"
      }
    },
```
and
```
    {
      "match": {
        "uri": "/netbox/static/*"
      },
      "action": {
        "share": "/opt/netbox/netbox/static"
      }
    },
```

**Comment by kkthxbye-code** [comment]
From looking at the documentation for nginx unit, it doesn't seem to support url rewriting yet, so my suggestion will not work with the netbox-docker image in most cases.

In your specific case, it seems like you want the base path to be /netbox. That might be possible like this:

```    
{
      "match": {
        "uri": "/netbox/static/*"
      },
      "action": {
        "share": "/opt/netbox"
      }
    },
```

This will pass all /netbox/static/* urls to /opt/share/netbox so if you request https://netboxhost.com/netbox/static/blah.js it will be fetched from /opt/netbox + /netbox/static/blah.js which should give the right URL.

Nginx unit seems pretty limited in general, so I don't have a good solution for you. It's not possible to just use a subdomain for your netbox instance instead of a path?

**Comment by kale1d0code** [comment]
Thanks, with your changes I can get into the application.
Although I could dedicate a subdomain for netbox, other people may not have that option. I've been using another nginx instance as a workaround.

I would imagine as nginx unit matures, it will be possible to use different urls

**Comment by Kani999** [comment]
The other workaround would be to copy the static files out of the docker container to host machine and set up the nginx at the host to:

- `app1` is the `BASE_PATH`
```
server {
        ...
        location /app1/static/ {
            alias /srv/netbox-docker/static/;
        }

        location /app1/ {
            proxy_pass http://127.0.0.1:8000;
            proxy_set_header X-Forwarded-Host $host:$server_port;
            proxy_set_header X-Forwarded-Server $host;
            proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        }
}
```

**Comment by ghost** [comment]
If you're using traefik as a frontend you can use this middleware as well, change the regex to a static value if it fits your use case (my use case has multiple netbox instances on the same domain name routed by host and path prefix)
```yaml
http:
  middlewares:
    netbox-static-fix:
      replacePathRegex:
        regex: "^/(.*)/static/"
        replacement: "/static/"
```

And on the service ` - traefik.http.routers.netbox-${NETBOX_STACK}.middlewares=netbox-static-fix@file`

**Comment by MarcHagen** [comment]
There is  a `UNIT_CONFIG` at https://github.com/netbox-community/netbox-docker/blob/58a157983259d3a9992776dbe9f166ca11d2f987/docker/launch-netbox.sh#L3

So you can pass your own config file.

**Comment by jphernandez1587** [comment]
Thanks to the comments on this issue I was able to come around this as follows (in case it might be helpful to someone else):
- The static content must be moved to a path of the form /opt/netbox/<BASE_PATH>/static inside the Netbox container, so I created a new Docker image based on the Netbox official image:

```
FROM netboxcommunity/netbox:latest-ldap

ARG BASE_PATH

RUN mkdir /opt/netbox/$BASE_PATH \
    && mv /opt/netbox/netbox/static /opt/netbox/$BASE_PATH/static
```

- And then in the docker-compose.yml I referenced my custom image and loaded a custom nginx-unit.json in the volumes section:

"my-nginx-unit.json:/etc/unit/nginx-unit.json:z,ro"

- The nginx-unit.json needs also to be adjusted accordingly:

```
"routes": [
  {
    "match": {
      "uri": "/{{ BASE_PATH }}/static/*"
    },
    "action": {
      "share": "/opt/netbox"
    }
  },
```

This way it is not necessary to use an external NGINX server and still rely on the community Netbox image. This should work for any base path, not only `netbox`

**Comment by candlerb** [comment]
This is now "closed as completed".

But I would like to clarify: to run Netbox with `BASE_PATH` do you still need the [workaround](https://github.com/netbox-community/netbox-docker/issues/650#issuecomment-994862043) from @jphernandez1587 - i.e. build your own image with the static directory moved?  Or can this now be done directly from the published image, and if so, how?

I notice there is no `BASE_PATH` in [nginx_unit.json](https://github.com/netbox-community/netbox-docker/blob/release/docker/nginx-unit.json) so this still needs to be overridden, but that's easy enough if that's all that's required, using a volume mount (in conjunction with `UNIT_CONFIG` if desired).

The question really boils down to: can nginx-unit strip off the `BASE_PATH` prefix from `$uri` for static files, e.g. using [njs scripting](https://unit.nginx.org/configuration/#scripting)?

For the special (but common) case where `BASE_PATH` is `netbox/`, I guess you could change:

```
    {
      "match": {
        "uri": "/static/*"
      },
      "action": {
        "share": "/opt/netbox/netbox${uri}"
      }
    },
```
to:
```
    {
      "match": {
        "uri": "/netbox/static/*"
      },
      "action": {
        "share": "/opt/netbox${uri}"
      }
    },
```
Ergh!

**Comment by monsdar** [comment]
Trying to go with what @candlerb posted. I'm not an expert in that matter, but what he writes makes sense to me. Unfortunately I cannot get it going. Any ideas what this approach is missing?

* Running with `netboxcommunity/netbox:v3.7.2-2.8.0`
* Within the image the following pathes exist:
  * `/opt/netbox` containing `pyproject.toml`, a few txt-files and the `netbox` subdirectory
  * `/opt/netbox/netbox` containing the `static` subdirectory and all the other netbox subdirs like `netbox`, `accounts`, ...
* Copied [nginx-unit.json](https://github.com/netbox-community/netbox-docker/blob/release/docker/nginx-unit.json) and made the change @candlerb proposed for my page running under `/netbox`.
  * This results in `"uri": "/netbox/static/*"` is configured to `"share": "/opt/netbox${uri}"` which should result in `/opt/netbox/netbox/static`. This is where all the static files are within the image.

What I'm getting with this solution is the following:
* I visit `https://my.domain.com/netbox`
* I get forwarded to `https://my.domain.com/media-failure/?filename=setmode.js` which isn't available because the `/netbox` is missing.

The other solutions from above seem not to work (anymore?). Creating my own image where I move `/opt/netbox` into `/opt/netbox/netbox` does not make much sense as there already is such a directory and all the files are there.

I don't think we're far from having an easy workaround/solution for this issue. Would be awesome if someone could chime in and point out what I'm missing right now...

**Comment by monsdar** [comment]
After wondering why the configured BASE_PATH hasn't been reflected within my netbox-instance I found that it has been removed from the configuration.py in this commit: https://github.com/netbox-community/netbox-docker/commit/155e90c99fc3743dab2ae9d9a5f105b66ca946fa

This is very confusing, as the docs still state that the parameter is available: https://demo.netbox.dev/static/docs/configuration/system/#base_path

As a user I'd assume at least a log line like "WARNING: You're using BASE_PATH which isn't supported by the Netbox Docker image".

**Comment by monsdar** [comment]
That was it... I applied the changed like I described in my [previous comment](https://github.com/netbox-community/netbox-docker/issues/650#issuecomment-1951425876) but also added the following line back into the configuration.py and mounted it into the container:

```
# Base URL path if accessing NetBox within a directory. For example, if installed at http://example.com/netbox/, set:
# BASE_PATH = 'netbox/'
BASE_PATH = environ.get('BASE_PATH', '')
```

and in my docker-compose.yml:

```
volumes:
      - ./nginx-unit.json:/etc/unit/nginx-unit.json:z,ro
      - ./configuration.py:/etc/netbox/config/configuration.py:z,ro
```

Hope this helps someone in the future...

**Comment by mlow** [comment]
Anybody else appearing here, you **do not** need to move or link the static assets to a different location within the container if your main goal is just to set `BASE_PATH`.

Below is a working `nginx-unit.json` snippet which is utilizing `substring` to strip the path prefix (not pretty at all, but probably necessary until Nginx Unit supports regex captures in rewrites):

```json
{
  "listeners": {
    "0.0.0.0:8080": {
      "pass": "routes/main"
    },
    "[::]:8080": {
      "pass": "routes/main"
    },
    "0.0.0.0:8081": {
      "pass": "routes/status"
    },
    "[::]:8081": {
      "pass": "routes/status"
    }
  },
  "routes": {
    "main": [
      {
        "match": {
          "uri": "/netbox/static/*"
        },
        "action": {
          "share": "`/opt/netbox/netbox${uri.substring(7)}`"
        }
      },
      {
        "action": {
          "pass": "applications/netbox"
        }
      }
    ],
    "status": [
      {
        "match": {
          "uri": "/netbox/status/*"
        },
        "action": {
          "proxy": "http://unix:/opt/unit/unit.sock"
        }
      }
    ]
  },
  "applications": {
    "netbox": {
      "type": "python 3",
      "path": "/opt/netbox/netbox/",
      "module": "netbox.wsgi",
      "home": "/opt/netbox/venv",
      "processes": {
        "max": 4,
        "spare": 1,
        "idle_timeout": 120
      }
    }
  },
  "access_log": "/dev/stdout"
}
```

(Above assumes `BASE_PATH = 'netbox/'`)

To recap, if you want to change `BASE_PATH`, you need to:
- Override `nginx-unit.json` similar to above (change the substr length to match your prefix)
- Put `BASE_PATH = 'your-path/` in `configuration/extra.py`
- Update the `netbox` container's health check to include your prefix.
